src/ClassicUO.Assets/FontsLoader.cs

```diff
@@ -44,6 +44,91 @@
 
 namespace ClassicUO.Assets
 {
+    public sealed class ASCIIFont
+    {
+        private const byte NOPRINT_CHARS = 32;
+        public readonly FontCharacterData[] characters = new FontCharacterData[224];
+
+        public unsafe IntPtr Load(IntPtr ptr, IntPtr end)
+        {
+            ptr += sizeof(ASCIIFontHeader);
+            if (ptr > end)
+            {
+                return ptr;
+            }
+
+            for (int j = 0; j < 224; j++)
+            {
+                var header = (ASCIIFontCharacterHeader *)ptr;
+                ptr = (IntPtr)(header + 1);
+                if (ptr > end)
+                {
+                    return ptr;
+                }
+
+                byte w = header->Width;
+                byte h = header->Height;
+
+                var data = (ushort *)ptr;
+                ptr = (IntPtr)(data + w * h);
+                if (ptr > end)
+                {
+                    return ptr;
+                }
+
+                characters[j] = new FontCharacterData(w, h, data);
+            }
+
+            return ptr;
+        }
+
+        /// <summary> Get the index in ASCII fonts of a character. </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static int GetIndex(char c)
+        {
+            byte ch = (byte)c; // ASCII fonts cover only 256 characters
+
+            if (ch < NOPRINT_CHARS)
+            {
+                return 0;
+            }
+
+            return ch - NOPRINT_CHARS;
+        }
+
+        public ref FontCharacterData GetCharacter(char c)
+        {
+            return ref characters[GetIndex(c)];
+        }
+
+        public int GetWidth(string str)
```

Comment:
FYI for future: Use `ReadOnlySpan<char>`, there is an implicit conversion. Doing a `for` loop over that is faster than the string itself because JIT will reach into the actual char array and make optimizations under the hood.

Reply:
That would be an optimization unrelated to this PR. But it's an obscure and weird DotNet implementation detail. In C++, iterating over a `std::string` and `std::span` generates the exact same machine code.

Reply:
Yeah, it's definitely a .net implementation detail and we should handle that in it's own iteration at a later time.